docs/templates/model-serving/predictor.md

```diff
@@ -0,0 +1,167 @@
+
+??? info "Show serving tools comparison"
+
+    | Feature / requirement                          | Docker       | Kubernetes (enterprise) | KServe (enterprise)         |
+    | ---------------------------------------------- | ------------ | ----------------------- | --------------------------- |
+    | Autoscaling (scale-out)                        | ❌           | ✅                      | ✅                          |
+    | Resource allocation                            | ➖ fixed     | ➖ min. resources       | ✅ min / max. resources     |
+    | Inference logging                              | ➖ simple    | ➖ simple               | ✅ fine-grained             |
+    | Inference batching                             | ➖ partially | ➖ partially            | ✅                          |
+    | Scale-to-zero                                  | ❌           | ❌                      | ✅ after 30s of inactivity) |
+    | Transformers                                   | ❌           | ❌                      | ✅                          |
+    | Low-latency predictions                        | ❌           | ❌                      | ✅                          |
+    | Multiple models                                | ❌           | ❌                      | ➖ (python-based)           |
+    | Custom predictor required <br /> (python-only) | ✅           | ✅                      | ❌                          |
+
+To learn how to specify the serving tool in a deployment, see [Serving Tool Reference](../predictor_api/#serving_tool).
+
+### Custom predictor
+
+Depending on the model server and serving tool used in the deployment, users can provide their own python script to load the model and make predictions.
+
+??? info "Show supported custom predictors"
+
+    | Serving tool | Model server       | Custom predictor script |
+    | ------------ | ------------------ | ----------------------- |
+    | Docker       | Flask              | ❌                      |
+    |              | Tensorflow Serving | ✅                      |
+    | Kubernetes   | Flask              | ❌                      |
+    |              | Tensorflow Serving | ✅                      |
+    | KServe       | Flask              | ✅                      |
+    |              | Tensorflow Serving | ✅                      |
+
+To configure a custom predictor, users must provide a python script implementing the following class.
+
+=== "Python"
+
+    ``` python
+    class Predict(object):
+
+        def __init__(self):
+            """ Initialization code goes here:
+                - Download the model artifact
+                - Load the model
+            """
+            pass
+
+        def predict(self, inputs):
+            """ Serve predictions using the trained model"""
+            pass
+    ```
+
+The path to this script has to be provided when calling `deploy()` or `create_predictor()` methods. Find more details in the [Predictor Reference](predictor_api.md).
```

Comment:
The predictor script should be available via a local file system path or a path on HopsFS. The path to this script then has to be provided...